--- a/libcxx/include/__config
+++ b/libcxx/include/__config
@@ -131,17 +131,21 @@
 //      Use rand_s(), for use on Windows.
 //      When this option is used, the token passed to `std::random_device`'s
 //      constructor *must* be "/dev/urandom" -- anything else is an error.
-#  if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) ||                     \
-      defined(__DragonFly__)
-#    define _LIBCPP_USING_ARC4_RANDOM
-#  elif defined(__wasi__) || defined(__EMSCRIPTEN__)
-#    define _LIBCPP_USING_GETENTROPY
-#  elif defined(__Fuchsia__)
-#    define _LIBCPP_USING_FUCHSIA_CPRNG
-#  elif defined(_LIBCPP_WIN32API)
-#    define _LIBCPP_USING_WIN32_RANDOM
-#  else
-#    define _LIBCPP_USING_DEV_RANDOM
+#  if !defined(_LIBCPP_USING_ARC4_RANDOM) && !defined(_LIBCPP_USING_GETENTROPY) && \
+      !defined(_LIBCPP_USING_DEV_RANDOM) && !defined(_LIBCPP_USING_FUCHSIA_CPRNG) && \
+      !defined(_LIBCPP_USING_WIN32_RANDOM)
+#    if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || \
+        defined(__DragonFly__)
+#      define _LIBCPP_USING_ARC4_RANDOM
+#    elif defined(__wasi__) || defined(__EMSCRIPTEN__)
+#      define _LIBCPP_USING_GETENTROPY
+#    elif defined(__Fuchsia__)
+#      define _LIBCPP_USING_FUCHSIA_CPRNG
+#    elif defined(_LIBCPP_WIN32API)
+#      define _LIBCPP_USING_WIN32_RANDOM
+#    else
+#      define _LIBCPP_USING_DEV_RANDOM
+#    endif
 #  endif
 
 #  ifndef _LIBCPP_CXX03_LANG
